src/core/cgroup.c

```
@@ -2204,12 +2204,6 @@ int unit_attach_pids_to_cgroup(Unit *u, Set *pids, const char *suffix_path) {
         if (set_isempty(pids))
                 return 0;
 
-        /* Load any custom firewall BPF programs here once to test if they are existing and actually loadable.
-         * Fail here early since later errors in the call chain unit_realize_cgroup to cgroup_context_apply are ignored. */
-        r = bpf_firewall_load_custom(u);
```

The idea behind this is that the custom bpf programs are local to the system, hence we should insist them being available if specified. And we document the existing behavior. Please just move the test below unit_realize_cgroup.

---

I cannot follow. The man page says errors will be ignored, but we do not.
https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#BPFProgram=type:program-path
> a specified program does not exist (i.e. has not been uploaded to the BPF subsystem of the kernel yet), it will not be installed but unit activation will continue (a warning will be printed to the logs).

---

I think too, these should go, BPF stuff is so far document as something we gracefully degrade on. We might want to look into optionally changing that, but I think @yuwata's patch is inline both with current docs, and with the rest of the codebase, hence I think this should go in.

---

No, this is about IPIngressFilterPath=

---

OK. Also updated the document that the failures are gracefully ignored.